pysparkling/utils.py

```diff
@@ -24,3 +28,18 @@ def next(self, separator=None):
         value = self.expression[:sep_pos]
         self.expression = self.expression[sep_pos + len(separator):]
```

Comment:
This PR also fixes a bug in Tokenizer.next() is called with a list: `len(separator)` is the size of the list and not the length of the found separator. Hence `self.expression` is altered in a incorrect way, leading to an unexpected behaviour when called once again,

This use case (call with a list and then another call) only appear in fs.hdfs so there is no side effect in other modules